### ISSUE TITLE
[Object Detection][Feature Request] Option to restore (detection) weights from pre-trained model for existing classes when fine-tuning on *additional* classes

### ISSUE THREAD

**Issue by netanel-s** [tensorflow/models#5553]
### System information
- **What is the top-level directory of the model you are using**: Object Detection
- **Have I written custom code**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.10.
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: CUDA 9
- **GPU model and memory**: 1080Ti, 12GB
- **Exact command to reproduce**:  Fine-tuning a model with additional classes (relative to pre-trained model).

### Describe the problem
Let's say we have a pre-trained model on COCO, and I want to fine-tune it to have **additional** classes. 
Now, since the number of classes has changed, then only the weights up to the last layer(s) are restored from the pre-trained model, and all the weights of the detection heads at the end are initialized. 
I (and others I talked to) would like to have an option where the weights of the existing classes would be restored too, and fine-tuning would then be of both the restored weights of the existing classes and the initialized weights of the additional classes.
It should be investigated whether it makes sense to restore **all** weights of existing classes, but it sure makes sense to do it for example to the bounding box regression weights.
"Finding out" that the class set has **additional** classes relative to pre-trained model can be done by the label map.

Thanks in advance.

**Comment by pkulzc** [comment]
Thanks for bringing this up! 
We've met similar situations, but in my option it may not make sense to partially load the weights. 

See this [image](http://cs231n.github.io/assets/cnn/convnet.jpeg) in Stanford [CS231](http://cs231n.github.io/convolutional-networks/). Each conv op is actually a set of filters + weighted activation. It's possible that a conv op that predicts 90 classes well has pretty different set of filters from a conv op that predicts 100 classes well. At least we don't which ones are good and which ones are bad. 
In this case, re-train the whole conv seems to be a better solution to me. Happy to discuss if you have different options. 

Also currently we don't have cycles for this, but if you are willing to implement and experiment with this feature, I'll be glad to review and merge it into our framework. Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
